File: 12ejercicio.js
/*
Enunciado: Dado un número, si es par se divide entre 2; si es impar se multiplica por 3 y suma 1. Repetir hasta llegar a 1.
Requerimientos: Usa while (!= 1), if-else para la paridad y los operadores aritméticos.
*/

let n = 27; // Cambia este valor para probar
console.log('Inicio:', n);

while (n != 1) {
	if (n % 2 === 0) {
		n = n / 2;
	} else {
		n = n * 3 + 1;
	}
	console.log(n);
}

File: 12ejercicio.py
"""Enunciado: Multiplica un número por 2 y por 4 utilizando solo operadores a nivel de bits.
Requerimientos: Usa un bucle for y el operador bitwise de desplazamiento a la izquierda <<."""

def multiplicar_por_dos_y_cuatro(numero):
    resultado_dos = numero << 1  
    resultado_cuatro = numero << 2  
    return resultado_dos, resultado_cuatro

numero = 5
resultado_dos, resultado_cuatro = multiplicar_por_dos_y_cuatro(numero)
print(f"{numero} multiplicado por 2 es: {resultado_dos}")  
print(f"{numero} multiplicado por 4 es: {resultado_cuatro}") 
